formats/json/commonTest/src/kotlinx/serialization/JsonPathTest.kt

```diff
@@ -32,6 +32,13 @@ class JsonPathTest : JsonTestBase() {
         expectPath("$.i.d[1]") { Json.decodeFromString<Outer>("""{"a":42, "i":{"d":{1:{}}""") }
     }
 
+    @Test
+    fun testUnknownKeyIsProperlyReported() {
+        expectPath("$.i") { Json.decodeFromString<Outer>("""{"a":42, "i":{"foo":42}""") }
+        expectPath("$") { Json.decodeFromString<Outer>("""{"x":{}, "a": 42}""") }
+//        Json.decodeFromString<Outer>("""{"a":42, "x":{}}""")
```

Comment:
This is the only place where misattribution happens, but I'm not sure how to fix it without a huge performance impact, so maybe we can live with that

Reply:
Please add this comment directly to the code